src/d3a/models/strategy/external_strategy.py

```diff
@@ -141,11 +156,38 @@ def _delete_offer(self, payload):
         except Exception:
             self.publish(
                 self._offer_response_channel,
-                {"error": "Incorrect offer request. Available parameters: (offer)."}
+                {"error": "Incorrect delete offer request. Available parameters: (offer)."}
             )
         else:
             return self._delete_offer_impl(arguments)
 
+    def _bid(self, payload):
+        try:
+            arguments = self._parse_payload(payload)
+            assert set(arguments.keys()) == {'price', 'energy'}
+            arguments['buyer'] = self.area.name
+            arguments['seller'] = self.area.parent.name
+        except Exception:
+            self.publish(
+                self._bid_response_channel,
+                {"error": "Incorrect offer request. Available parameters: (price, energy)."}
+            )
+        else:
+            return self._bid_impl(arguments)
+
+    def _delete_bid(self, payload):
+        try:
+            arguments = self._parse_payload(payload)
+            assert set(arguments.keys()) == {'bid'}
+            arguments['bid_or_id'] = arguments.pop('bid')
+        except Exception:
+            self.publish(
+                self._delete_bid_response_channel,
+                {"error": "Incorrect offer request. Available parameters: (bid)."}
```

Comment:
Very minor but you could rename "Incorrect offer request" to "Incorrect bid request"  (also in _bid).

Reply:
Edit: Here you can even use "delete_offer request"

Reply:
Well, there are different parameters for offer and bid, therefore we cannot reuse the same method. Adapted the offer -> bid rename though

Reply:
Ah sorry, i misspelled. I meant "delete_bid request.

Reply:
Ahhh I see now, thanks a lot. Will fix this